### PR TITLE
Make `get` script independent from `make`

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
We now do not rely on `make` any more for sake of a higher portability
of the `get` script.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: scripts/get [hunk 1]
```diff
@@ -94,8 +94,54 @@ download_and_install() {
...
+    CNIDIR=$ETCDIR/cni/net.d
+    BINDIR=$PREFIX/bin
+    MANDIR=$PREFIX/share/man
+    OCIDIR=$PREFIX/share/oci-umount/oci-umount.d
+    SELINUX=$(selinuxenabled 2>/dev/null && echo -Z)
+    BASHINSTALLDIR=$PREFIX/share/bash-completion/completions
+    FISHINSTALLDIR=$PREFIX/share/fish/completions
+    ZSHINSTALLDIR=$PREFIX/share/zsh/site-functions
+    SYSTEMDDIR=$PREFIX/lib/systemd/system
+    OPT_CNI_BIN_DIR=/opt/cni/bin
```
Comment (haircommander) [inline scripts/get:111]: is there any chance we can have a shared source file for these, in case the makefile values change?

Comment (saschagrunert) [reply scripts/get:111]: Hm, we could ship two additional scripts: `install` and `uninstall`. The `install` script could be re-used by the `get` script, but we would have to invoke an additional `curl` command... Not sure about how to solve it in a cleaner way...

File: scripts/get [hunk 2]
```diff
@@ -94,8 +94,54 @@ download_and_install() {
...
+
+    install "$SELINUX" -d -m 755 $CNIDIR
```
Comment (harche) [inline scripts/get:113]: Thanks, I just verified `install` command exists on FCOS by default. It ships with very minimal set of commands, so I thought I will make sure anyway. 

```bash
[root@localhost core]# which install 
/usr/bin/install
```

File: scripts/get (hunk 1 above)
Comment (kolyshkin) [reply scripts/get:111]: We can have something like `scripts/install-paths`:
```make
# NOTE this file is sourced from make and shell
PREFIX=/usr/local
ETCDIR=/etc
CONTAINERS_DIR=$ETCDIR/containers
....
```

and we can use this from Makefile:
```make
include ./scripts/install-paths

...

install: 
        install ${SELINUXOPT} -D -m 755 bin/crio $(DESTDIR)$(BINDIR)/crio
.....
```

as well as from the `scripts/get`:
```sh
. install-paths
```

That way we at least have a single source for all these paths.

NOTE that currently `DESTDIR` is not handled correctly by the Makefile. It should never be included into `PREFIX`, but rather specified directly as in my example above.

Comment (kolyshkin) [reply scripts/get:111]: > and uninstall

I would drop the `uninstall` target from the makefile. It is not very robust anyway, as no one keeps track of what's installed. For example, `make DESTDIR=$(pwd)/inst install` followed by `make PREFIX=/usr uninstall` might do some real harm.

Comment (saschagrunert) [reply scripts/get]: Removed the uninstall part from the makefile and changed the implementation to just provide an `install` script.

**Comment by saschagrunert** [comment]
Added some GitHub actions test case for the code-generation part. This should now work, PTAL again @haircommander @kolyshkin

File: scripts/get [hunk 3]
```diff
@@ -83,26 +83,77 @@ latest_version() {
...
+
+# In this section we include the contents of
+# contrib/bundle/install-paths and contrib/bundle/install
+
+# INCLUDE
+DESTDIR=${DESTDIR:-}
+PREFIX=${PREFIX:-/usr/local}
+ETCDIR=${ETCDIR:-/etc}
+CONTAINERS_DIR=${CONTAINERS_DIR:-$ETCDIR/containers}
+CNIDIR=${CNIDIR:-$ETCDIR/cni/net.d}
```
Comment (haircommander) [inline scripts/get:114]: are we ablet o source the contrib scripts here?

Comment (saschagrunert) [reply scripts/get:114]: We pipe the script through curl, which means we would have to `curl` again at this point. Do we wanna do that or stick to the current `make get-scripts` approach?

Comment (haircommander) [reply scripts/get:114]: ach, damn, of course. No that's fine. I'd like some way that we future proof these fields, so they're not inconsitent between the (now two) places they exist (top-level makefile and here). not blocking on this PR, but probably some auto generation and hack/tree_status.sh will be the method...

Comment (saschagrunert) [reply scripts/get:114]: We have it via `make get-script`, the new GitHub actions job will verify that the content is sane:
https://github.com/cri-o/cri-o/pull/4610/checks?check_run_id=2022361523

:)

Comment (haircommander) [reply scripts/get:114]: then the only thing that needs fixing is my ability to read :)

**Comment by haircommander** [comment]
/approve

LGTM

**Review by harche — APPROVED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM 
assuming happy tests

**Comment by saschagrunert** [comment]
@harche I think the script is now ready to be used within the k8s CI :)

**Pull request merged**